packages/orchestrator/internal/proxy/proxy.go

```diff
@@ -0,0 +1,193 @@
+package proxy
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/e2b-dev/infra/packages/shared/pkg/meters"
+	"github.com/e2b-dev/infra/packages/shared/pkg/smap"
+	"go.uber.org/zap"
+	"html/template"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+//go:embed proxy_browser_502.html
+var proxyBrowser502PageHtml string
+
+var browserIdentityKeywords = []string{
+	"mozilla", "chrome", "safari", "firefox", "edge", "opera", "msie",
+}
```

Comment:
feel free to generate the regex based on the current slice, it's easier to read
```suggestion
var browserRegex = regexp.MustCompile(`(?i)mozilla|chrome|safari|firefox|edge|opera|msie`)
```